### PR TITLE
Added DIPSColor markup extension

### PR DESCRIPTION
**Pull request by haavamoa**
## Added / Fixed

- Added a new feature

## Usage:
```xml
<BoxView BackgroundColor="{dxui:DIPSColor Theme=TealPrimary}" />
<BoxView BackgroundColor="{dxui:DIPSColor ColorPalette=DarkAir}"/>
<BoxView BackgroundColor="{dxui:DIPSColor StatusColorPalette=Danger}"/>
<BoxView BackgroundColor="{dxui:DIPSColor StatusColorPalette=WarningLight}"/>
```

## Todo List

- [X] I have added unit tests
- [X] I have tested on an Android device.
- [X] I have tested on an iOS device.

### CODE REVIEW

**Comment by haavamoa** [comment]
Identifiers are nullable to be able to know if / how many colors have been tried to set, I might be able to use `IServiceProvider` to know that and that means I can remove the nullable.

File: src/DIPS.Xamarin.UI/Extensions/Markup/DIPSColorExtension.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            return ProvideValue(serviceProvider);
+        }
+
+        /// <inheritdoc/>
+        public Color ProvideValue(IServiceProvider serviceProvider)
+        {
+            var listOfBools = new List<bool>() { Theme != null, StatusColorPalette != null, ColorPalette != null };
+            Tuple<string, string?> xamlInfo;
+
+            if (listOfBools.Count(b => b) > 1)//More than one color family is set
```
Comment (Illedan) [inline src/DIPS.Xamarin.UI/Extensions/Markup/DIPSColorExtension.cs:37]: This feels like magic

Comment (haavamoa) [reply src/DIPS.Xamarin.UI/Extensions/Markup/DIPSColorExtension.cs:37]: What exactly?

File: src/DIPS.Xamarin.UI/Extensions/Markup/DIPSColorExtension.cs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        }
+
+        /// <inheritdoc cref="ColorPalette.Identifier"/>
+        public ColorPalette.Identifier ColorPalette
+        {
+            get => m_colorPalette;
+            set
+            {
+                m_colorPalette = value;
+                m_colorPaletteWasSet = true;
```
Comment (Illedan) [inline src/DIPS.Xamarin.UI/Extensions/Markup/DIPSColorExtension.cs:52]: Can be set to null?

File: src/DIPS.Xamarin.UI/Extensions/Markup/DIPSColorExtension.cs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        private bool m_colorPaletteWasSet;
+
+        /// <inheritdoc cref="Theme.Identifier"/>
+        public Theme.Identifier Theme
+        {
+            get => m_theme;
+            set
+            {
+                m_theme = value;
+                m_themeWasSet = true;
```
[the remark this answers is not in the document]
Comment (haavamoa) [reply src/DIPS.Xamarin.UI/Extensions/Markup/DIPSColorExtension.cs:30]: It is not possible, the property will have to be `nullable` for that to be possible.

**Pull request merged**